core/src/main/java/com/amplifyframework/devmenu/EnvironmentInfo.java

```diff
@@ -0,0 +1,116 @@
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.AmplifyException;
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.Resources;
+import com.amplifyframework.core.category.Category;
+import com.amplifyframework.core.plugin.Plugin;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Contains information about the plugin versions added to the application
+ * and the developer's environment.
+ */
+public final class EnvironmentInfo {
+    // Name of the file that contains the developer environment information.
+    private static final String DEV_ENV_INFO_FILE_NAME = "localenvinfo";
+    // A map from developer environment information items to the formatted name of that item.
+    private final Map<String, String> devEnvironmentItems;
+
+    /**
+     * Constructs a new EnvironmentInfo instance.
+     */
+    public EnvironmentInfo() {
+        devEnvironmentItems = new HashMap<>();
+        devEnvironmentItems.put("nodejsVersion", "Node.js Version");
+        devEnvironmentItems.put("npmVersion", "NPM Version");
+        devEnvironmentItems.put("amplifyCliVersion", "Amplify CLI Version");
+        devEnvironmentItems.put("androidStudioVersion", "Android Studio Version");
+        devEnvironmentItems.put("osName", "OS");
+        devEnvironmentItems.put("osVersion", "OS Version");
+    }
+
+    /**
+     * Returns a String representation of the version of each plugin added, or
+     * "No plugins added." if no plugins were added.
+     * @return version information for each plugin added
+     */
+    public String getPluginVersions() {
+        String pluginVersions = getCategoryPluginVersions(Amplify.Analytics) +
+                getCategoryPluginVersions(Amplify.API) +
+                getCategoryPluginVersions(Amplify.Auth) +
+                getCategoryPluginVersions(Amplify.Logging) +
+                getCategoryPluginVersions(Amplify.Storage) +
+                getCategoryPluginVersions(Amplify.Hub) +
+                getCategoryPluginVersions(Amplify.DataStore) +
+                getCategoryPluginVersions(Amplify.Predictions);
+        return pluginVersions.isEmpty() ? "No plugins added." : pluginVersions;
+    }
+
+    /**
+     * Returns a String representation of information about the developer's environment.
+     * @param context an Android Context
+     * @return developer environment information
+     * @throws AmplifyException if the developer environment information could not be read
+     */
+    public String getDeveloperEnvironmentInfo(@NonNull Context context) throws AmplifyException {
+        Context appContext = Objects.requireNonNull(context).getApplicationContext();
+        JSONObject envInfo = Resources.readJsonResource(appContext, DEV_ENV_INFO_FILE_NAME);
+        StringBuilder formattedEnvInfo = new StringBuilder();
```

Comment:
Let's assume this info will be in `amplifyconfiguraiton.json`, pending some support from the CLI team. If that's so, you can write this in a forward-compatible way by checking:
```java
if (amplifyConfigJson.has("keyName")) {
    // use it!
} else {
    // env info not available
}
```